tests/test_sensor.py

```diff
@@ -80,6 +86,87 @@ def status_func(value):
     assert sensor.status == Sensor.Status.NOMINAL
 
 
+class MyEnum(enum.Enum):
+    ZERO = 0
+    ONE = 1
+
+
+class OtherEnum(enum.Enum):
+    ABC = 0
+    DEF = 1
+
+
+@pytest.mark.parametrize(
+    "sensor_type,compatible_value",
+    [
+        (bool, True),
+        (int, 1_2_3_4),
```

Comment:
I couldn't find another subclass/compatible type of `numbers.Integral`. This syntax (new to me) should be handled by Python by default, so might not be the most rigorous test.

Reply:
I wouldn't both with that syntax. It makes no difference to the actual value/type.

I can't think of any other built-in concrete Integral classes either. Let's just test the float/Real case (with Fraction) and assume that Integral will work similarly.